lib/dry/transaction/step_failure.rb

```diff
@@ -12,6 +12,10 @@ def method_missing(name, *args, &block)
         @__object.send(name, *args, &block)
       end
 
+      def respond_to_missing?(*args)
+        @__object.respond_to?(*args)
```

Comment:
This is missing `|| super` and the `include_private` argument should default to false, also need to pass `true` as the second argument to `respond_to?` as it's using `send` rather than `public_send`